zio-http/src/main/scala/zhttp/service/Server.scala

```diff
@@ -34,21 +36,25 @@ object Server {
     port: Int = 8080,
     leakDetectionLevel: LeakDetectionLevel = LeakDetectionLevel.SIMPLE,
     maxRequestSize: Int = 4 * 1024, // 4 kilo bytes
+    maybeSslContext: Option[JSslContext] = None,
   )
 
   private case class Concat[R, E](self: Server[R, E], other: Server[R, E]) extends Server[R, E]
   private case class Port(port: Int)                                       extends UServer
   private case class LeakDetection(level: LeakDetectionLevel)              extends UServer
   private case class MaxRequestSize(size: Int)                             extends UServer
+  private case class SslContext(context: JSslContext)                      extends UServer
   private case class App[R, E](http: Http[R, E])                           extends Server[R, E]
 
   def app[R, E](http: Http[R, E]): Server[R, E] = Server.App(http)
   def maxRequestSize(size: Int): UServer        = Server.MaxRequestSize(size)
   def port(int: Int): UServer                   = Server.Port(int)
-  val disableLeakDetection: UServer             = LeakDetection(LeakDetectionLevel.DISABLED)
-  val simpleLeakDetection: UServer              = LeakDetection(LeakDetectionLevel.SIMPLE)
-  val advancedLeakDetection: UServer            = LeakDetection(LeakDetectionLevel.ADVANCED)
-  val paranoidLeakDetection: UServer            = LeakDetection(LeakDetectionLevel.PARANOID)
+  def ssl(context: JSslContext): UServer        = Server.SslContext(context)
```

Comment:
Instead of using netty's `SSLContext` we should create a more expressive API for users to interact with.